src/bp/nlu/stan/model-repo.ts

```diff
@@ -65,16 +123,52 @@ export default class ModelRepository {
       ['model']
     )
     const buffer = await fse.readFile(archiveName)
-    const fpath = path.join(modelDir, modelFileName)
-    await fse.writeFile(fpath, buffer)
+    await scopedGhost.upsertFile(MODELS_DIR, modelName, buffer)
     tmpDir.removeCallback()
   }
 
-  private _makeFileName(modelId: NLUEngine.ModelId, password: string): string {
+  // // implement this properly
+  // is this going top be publicly acessible over http ?
+
+  // currently hard to achieve a secure list model,
+  // we can't assure that the models are properly salted with the appSecret
+  // in other words, anyone could list models corresponding to an appId without having the secret
+
+  // potential solution
+  // we could come up with a signature that is created by a combination of appID+appSecret that is appended at the end of the model id
+  // this way we could validate the combination straight from the ghost using the file ending pattern
+  public async listModels(options: ModelOwnershipOptions) {
+    if (!options.appId) {
+      throw Error('cannot list models without owner')
```

Comment:
well, yes you can, its the root folder right ?

Reply:
I mean you can but we don't want to offer that do we? We can if there's some security preventing that at the API level.

If we don't add any security at the api level this means a random person can list all models at the root, potentially delete them